build.rs

```diff
@@ -15,10 +15,12 @@ fn main() {
         .replace("-", "_");
     println!("cargo::rerun-if-env-changed=MYSQLCLIENT_VERSION");
     println!("cargo::rerun-if-env-changed=MYSQLCLIENT_LIB");
+    println!("cargo::rerun-if-env-changed=MYSQLCLIENT_LIBNAME");
     println!("cargo::rerun-if-env-changed=MYSQLCLIENT_STATIC");
     println!("cargo::rerun-if-env-changed=MYSQLCLIENT_VERSION_{target}");
     println!("cargo::rerun-if-env-changed=MYSQLCLIENT_LIB_{target}");
     println!("cargo::rerun-if-env-changed=MYSQLCLIENT_STATIC_{target}");
+    let libname = env::var("MYSQLCLIENT_LIBNAME").unwrap_or("mysqlclient".to_string());
```

Comment:
This environment variable needs to be documented in the readme.

We also want to have a `MYSQLCLIENT_LIBNAME_{target}` variant of this as well to allow providing a target specific configuration.